File: apps/ios/Packages/AMUXUI/Sources/AMUXUI/Shared/HaiSheet.swift
import SwiftUI
import AMUXSharedUI

/// Small uppercase section header used to label a paper card on a Hai-style
/// sheet (pebble background, paper cards). See `IdeaSheet.swift` for the
/// canonical example.
struct HaiSectionLabel: View {
    let title: String
    init(_ title: String) { self.title = title }
    var body: some View {
        Text(title.uppercased())
            .font(.system(size: 11, weight: .semibold))
            .tracking(0.6)
            .foregroundStyle(Color.amux.basalt.opacity(0.7))
            .padding(.horizontal, 24)
    }
}

/// Reusable Hai sheet row body — left label, right value, optional chevron.
/// Plain layout (no surrounding card) so callers can stack multiple in one
/// paper card with hairline dividers.
struct HaiSheetRow: View {
    let label: String
    let value: String?
    let valueIsMonospaced: Bool
    let valueIsMuted: Bool
    let showsChevron: Bool

    init(label: String,
         value: String? = nil,
         valueIsMonospaced: Bool = false,
         valueIsMuted: Bool = false,
         showsChevron: Bool = false) {
        self.label = label
        self.value = value
        self.valueIsMonospaced = valueIsMonospaced
        self.valueIsMuted = valueIsMuted
        self.showsChevron = showsChevron
    }

    var body: some View {
        HStack(spacing: 8) {
            Text(label)
                .font(.system(size: 14.5))
                .foregroundStyle(Color.amux.onyx)
            Spacer(minLength: 8)
            if let value {
                Text(value)
                    .font(.system(size: 14, design: valueIsMonospaced ? .monospaced : .default))
                    .foregroundStyle(valueIsMuted ? Color.amux.basalt.opacity(0.6) : Color.amux.basalt)
                    .lineLimit(1)
                    .truncationMode(.middle)
            }
            if showsChevron {
                Image(systemName: "chevron.right")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundStyle(Color.amux.slate)
            }
        }
        .padding(.horizontal, 14)
        .padding(.vertical, 13)
    }
}

/// Wraps content in the standard Hai paper card (radius 14, paper fill,
/// horizontal inset matching the section label).
struct HaiPaperCard<Content: View>: View {
    @ViewBuilder var content: () -> Content

    var body: some View {
        VStack(spacing: 0) { content() }
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(
                RoundedRectangle(cornerRadius: 14, style: .continuous)
                    .fill(Color.amux.paper)
            )
            .padding(.horizontal, 16)
    }
}

File: apps/android/core/design/src/main/kotlin/tech/teamclaw/android/core/design/HaiSheet.kt
package tech.teamclaw.android.core.design

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Hai sheet primitives — Kotlin/Compose port of iOS `Shared/HaiSheet.swift`.
 *
 *   Pebble sheet background  →  HaiPaperCard (paper fill) on top
 *   Section label (uppercase, tracked) precedes each card
 *   HaiSheetRow stacks inside the card with HaiHairline dividers between rows
 *
 * Cinnabar is reserved for active CTAs / unread dots — never used here.
 */

@Composable
fun HaiSectionLabel(
    title: String,
    modifier: Modifier = Modifier,
) {
    Text(
        text = title.uppercase(),
        modifier = modifier.padding(horizontal = 24.dp),
        color = Hai.Basalt.copy(alpha = 0.70f),
        fontFamily = FontFamily.SansSerif,
        fontWeight = FontWeight.SemiBold,
        fontSize = 11.sp,
        letterSpacing = 0.6.sp,
    )
}

/**
 * Standard Hai paper card: 14dp radius, Hai.Paper fill, 16dp horizontal inset.
 * Stack [HaiSheetRow]s inside this card; insert [HaiHairline] between rows.
 */
@Composable
fun HaiPaperCard(
    modifier: Modifier = Modifier,
    content: @Composable () -> Unit,
) {
    Column(
        modifier = modifier
            .padding(horizontal = 16.dp)
            .fillMaxWidth()
            .clip(RoundedCornerShape(14.dp))
            .background(Hai.Paper),
    ) { content() }
}

/**
 * Reusable Hai sheet row: left label, right value, optional chevron.
 */
@Composable
fun HaiSheetRow(
    label: String,
    modifier: Modifier = Modifier,
    value: String? = null,
    valueIsMonospaced: Boolean = false,
    valueIsMuted: Boolean = false,
    showsChevron: Boolean = false,
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .padding(horizontal = 14.dp, vertical = 13.dp),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(8.dp),
    ) {
        Text(
            text = label,
            color = Hai.Onyx,
            fontFamily = FontFamily.SansSerif,
            fontWeight = FontWeight.Normal,
            fontSize = 14.5.sp,
        )
        Spacer(Modifier.weight(1f))
        Spacer(Modifier.width(8.dp))
        if (value != null) {
            Text(
                text = value,
                color = if (valueIsMuted) Hai.Basalt.copy(alpha = 0.6f) else Hai.Basalt,
                fontFamily = if (valueIsMonospaced) FontFamily.Monospace else FontFamily.SansSerif,
                fontWeight = FontWeight.Normal,
                fontSize = 14.sp,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
        if (showsChevron) {
            Text(
                text = "›",
                color = Hai.Slate,
                fontFamily = FontFamily.SansSerif,
                fontWeight = FontWeight.SemiBold,
                fontSize = 16.sp,
            )
        }
    }
}

/**
 * Hairline divider — 0.5dp Hai.Hairline. Use between [HaiSheetRow]s.
 */
@Composable
fun HaiHairline(modifier: Modifier = Modifier) {
    HorizontalDivider(
        modifier = modifier.padding(horizontal = 14.dp),
        thickness = 0.5.dp,
        color = Hai.Hairline,
    )
}
